staging/src/k8s.io/sample-controller/artifacts/examples/main.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//This is an example of polymorphic scale client.
+
+package main
+
+import (
+	"flag"
+	"os"
+
+	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/clientcmd"
+	clientset "k8s.io/sample-controller/pkg/client/clientset/versioned"
+)
+
+const (
+	Namespace       = "default"
+	ResourceName    = "example-foo"
```

Comment:
Can do this without explicitly stating the name and namespace:

https://github.com/kubernetes/kubernetes/blob/8df7e8591e8cd7c7f61cf8c753f108503f4c0d09/staging/src/k8s.io/sample-controller/controller.go#L247-L248